### DISCUSSION TITLE
Is there limit on number of ATTACHed databases?

### DISCUSSION THREAD

**Discussion by lupko** [duckdb/duckdb#12214] [Q&A]
Hello all,

I wonder, is there any hard or practical limit to how many DuckDB databases can one ATTACH under one 'umbrella' DuckDB database?

For example `sqlite` which can also do ATTACH has some [limits](https://www.sqlite.org/limits.html) in this area. I was unable to find anything in the DuckDB docs. 

I did a quick experiment trying to attach up to 512 databases (each db had a single non-empty table in it, tried ro/rw mode). This gives me some indication that there may be no limit. Is that correct?

Also I wonder, are there any practical limits / considerations when using the ATTACH functionality? I anticipate using ATTACH in READ_ONLY mode in order to get up to 100 DuckDB database files under one umbrella DB. Is this something that DuckDB will handle with grace, or should I get ready for some 'funny stuff'?

Thanks,
Lubo

**Comment by Tishj** [comment]
Nothing as limiting as the 10-125 of sqlite

See <https://github.com/duckdb/duckdb/pull/9671> for an example, attaching 30k concurrently
